src/components/app/Home.js

```diff
@@ -78,6 +102,44 @@ class UploadButton extends React.PureComponent<UploadButtonProps> {
           />
           Select a profile to open
         </label>
+        <label
+          id="LoadFromUrlButton"
+          className="homeSectionButton"
+          onClick={this._loadFromUrlPressed}
+        >
+          Load profile from an URL
+        </label>
+        {this.state.isLoadFromUrlPressed ? (
+          <UploadFromUrl {...this.props} />
+        ) : null}
+      </div>
+    );
+  }
+}
+
+class UploadFromUrl extends React.PureComponent<UploadFromUrlProps> {
+  _input: HTMLInputElement | null;
+
+  _takeInputRef = input => {
+    this._input = input;
+  };
```

Comment:
instead of using the input ref, in React we use what we call "controlled input" instead. You can read about it in https://reactjs.org/docs/forms.html, but in short, the idea is that there's a local state that keeps the value, and we reuse this value in the `render`, and in the `_upload` method below.